src/utils/signer.ts

```diff
@@ -0,0 +1,440 @@
+/*
+ * This is a copied and modified file from https://github.com/aws-amplify/amplify-js/blob/main/packages/core/src/Signer/Signer.ts
+ * as well as several classes from https://github.com/aws-amplify/amplify-js/blob/main/packages/core/src/clients/middleware/signing/signer/signatureV4
```

Comment:
I'm assuming these are TOKEN_QUERY_PARAM, presignUrl, signRequest. Is that correct?

Reply:
Actually... I don't see signRequest copied over, which I do see it imported in https://github.com/aws-amplify/amplify-js/blob/main/packages/core/src/Signer/Signer.ts#L7. Is it needed?

Reply:
`signRequest` is only needed for `Signer.sign`, so I didn't copy it over since I only included the `Signer.signUrl` method.